toolkit/scripts/download-packages.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash -e
+
+function get_packages {
+    # First argument is an URL to a directory containing packages
+    download_url="$1"
+
+    # wget -nv -O - "$download_url"               -- Download HTML of package directory and send to stdout (-nv reduces verbosity)
+    # | grep 'a href'                             -- Filter out lines that are not links (we're looking for links to rpms)
+    # | sed -E -e 's:<a href="(.+[^\])".+:\1:'    -- Remove everything but the address part of links (these are already URL encoded)
+    # | grep -v '/'                               -- '/' is invalid in RPM name but appears in links to different directories. Filter these out.
+    # | xargs -I {} wget -nv "$download_url"/{}   -- Download the packages
+
+    echo "-- Downloading packages from $download_url."
+    SECONDS=0
+    wget -nv -O - "$download_url" | grep 'a href' | sed -E -e 's:<a href="(.+[^\])".+:\1:' | grep -v '/' | xargs -P8 -I {} wget -nv "$download_url"/{}
+    echo "-- Finished downloading packages from $download_url. Operation took $SECONDS seconds."
```

Comment:
Wouldn't this work:

```
wget -r -l1 --no-parent -A "*.rpm" [https://packages.microsoft.com/cbl-mariner/1.0/prod/update/x86_64/rpms](https://packages.microsoft.com/cbl-mariner/1.0/prod/update/x86_64/rpms)
```

I found it [here](https://unix.stackexchange.com/questions/117988/wget-with-wildcards-in-http-downloads). It downloaded all RPMs for me, the only overhead was `wget` reconstructing the whole directory structure (I'm guessing this also gets rid of the risk of collisions), but maybe there's another switch for that. And if there's none, it shouldn't be to much work to get everything in one directory.